### PR TITLE
List Basic information on available topologies

### PR DESCRIPTION
**Pull request by aarondees-blabs**
Added functionality to return a list of basic information on all available topologies.

Change Log:
 - ListTopologiesCreateRequest struct added
 - BasicTopologyInfo struct added
 - GetAllTopologies() function added, returns an array of
 BasicTopologyInfo

### CODE REVIEW

File: alien4cloud/alien4cloud_structs.go [hunk 1]
```diff
@@ -329,6 +329,12 @@ type DeploymentArtifact struct {
 	Description          string                 `json:"description,omitempty"`
 }
 
+type ListTopologiesCreateRequest struct {
```
Comment (laurentganne) [inline alien4cloud/alien4cloud_structs.go]: This new type may not be needed as the type searchRequest described above is identical

File: alien4cloud/topology.go [hunk 2]
```diff
@@ -517,3 +518,55 @@ func (t *topologyService) SaveA4CTopology(ctx context.Context, a4cCtx *TopologyE
...
+		return nil, errors.Wrap(err, "Cannot marshal an a4cgetTopologiesCreateRequest structure")
+	}
+
+	response, err := t.client.doWithContext(ctx,
+		"POST",
+		fmt.Sprintf("%s/catalog/topologies/search", a4CRestAPIPrefix),
+		[]byte(string(getTopoJSON)),
+		[]Header{
+			{
+				"Content-Type",
```
Comment (laurentganne) [inline alien4cloud/topology.go]: Loïc introduced recently constants for headers, you could use:
```
[]Header{contentTypeAppJSONHeader, acceptAppJSONHeader},
```

File: alien4cloud/topology.go [hunk 3]
```diff
@@ -517,3 +518,55 @@ func (t *topologyService) SaveA4CTopology(ctx context.Context, a4cCtx *TopologyE
 
 	return processA4CResponse(response, nil, http.StatusOK)
 }
+
+func (t *topologyService) GetAllTopologies(ctx context.Context) ([]BasicTopologyInfo, error) {
```
Comment (laurentganne) [inline alien4cloud/topology.go]: To have a more generic API similar to the REST API, we could name this function `GetTopologies()` and add the argument `query string` that you would pass below in the searchRequest.query parameter.

File: alien4cloud/alien4cloud_structs.go (hunk 1 above)
Comment (aarondees-blabs) [reply alien4cloud/alien4cloud_structs.go]: Yes, I agree. Done.

File: alien4cloud/topology.go (hunk 2 above)
Comment (aarondees-blabs) [reply alien4cloud/topology.go]: Done.

File: alien4cloud/topology.go (hunk 3 above)
Comment (aarondees-blabs) [reply alien4cloud/topology.go]: Sounds good, done.

**Review by laurentganne — APPROVED** [review]
Looks good!

**Pull request merged**